modules/@apostrophecms/ui/ui/apos/lib/tooltip.js

```diff
@@ -0,0 +1,13 @@
+export default {
+  updateOptions (tooltipObject) {
+    tooltipObject.options.defaultOffset = '11';
+    tooltipObject.options.defaultPlacement = 'bottom-end';
+    tooltipObject.options.defaultTemplate = `
+      <div class="apos-tooltip" role="tooltip">
+        <div class="apos-tooltip__arrow"></div>
+        <div class="apos-tooltip__inner"></div>
+      </div>`;
+    tooltipObject.options.defaultArrowSelector = '.apos-tooltip__arrow';
+    tooltipObject.options.defaultInnerSelector = '.apos-tooltip__inner';
+  }
+};
```

Comment:
Why avoid creating a component out of this? it has configuration, template, CSS .. I imagine if it were a component that was imported by others we wouldn't need the duplicate `Vue.directive` registrations, etc.

naive?

Reply:
Because it's not a component. v-tooltip does this using a custom directive, not a component, so the directive needs to be set up on every Vue app where it might appear. One downside of Apostrophe using multiple apps for various parts is that it needs to be set up on each one individually.